code/modules/clothing/glasses/hud.dm

```diff
@@ -102,6 +102,44 @@
 	hud_trait = TRAIT_SECURITY_HUD
 	glass_colour_type = /datum/client_colour/glass_colour/red
 
+/obj/item/clothing/glasses/hud/security/equipped(mob/living/carbon/human/user, slot)
+	. = ..()
+	if (!.)
+		return FALSE
+	RegisterSignal(user, COMSIG_MOB_MIDDLECLICKON, .proc/create_ping)
+
+/obj/item/clothing/glasses/hud/security/proc/create_ping(datum/source, atom/A, params)
+	var/list/modifiers = params2list(params)
+	var/mob/living/L = source
+	if(L.stat != CONSCIOUS)
+		return
+	if (!LAZYACCESS(modifiers, ALT_CLICK) && !LAZYACCESS(modifiers, CTRL_CLICK))
+		return
+	var/image/holder = L.hud_list[SEC_PING]
+	// not ideal, but it works.
+	addtimer(CALLBACK(src, .proc/remove_ping, holder), 10 SECONDS, TIMER_UNIQUE | TIMER_OVERRIDE)
+	if (LAZYACCESS(modifiers, ALT_CLICK))
+		var/mutable_appearance/MA = new /mutable_appearance()
+		MA.icon = 'icons/effects/effects.dmi'
+		MA.icon_state = "squestion"
+		MA.layer = ABOVE_OPEN_TURF_LAYER
+		MA.plane = 0
+		holder.appearance = MA
+		holder.loc = get_turf(A)
+		return COMSIG_MOB_CANCEL_CLICKON
+	else if (LAZYACCESS(modifiers, CTRL_CLICK))
+		var/mutable_appearance/MA = new /mutable_appearance()
+		MA.icon = 'icons/effects/effects.dmi'
+		MA.icon_state = "salert"
+		MA.layer = ABOVE_OPEN_TURF_LAYER
+		MA.plane = 0
+		holder.appearance = MA
+		holder.loc = get_turf(A)
+		return COMSIG_MOB_CANCEL_CLICKON
+
+/obj/item/clothing/glasses/hud/security/proc/remove_ping(image/holder)
+	holder.loc = null
```

Comment:
remember to qdel the image

Reply:
the image is created by HUD code and held by the mob.